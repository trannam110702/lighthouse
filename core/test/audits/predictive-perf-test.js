/**
 * @license
 * Copyright 2017 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import PredictivePerf from '../../audits/predictive-perf.js';
import {getURLArtifactFromDevtoolsLog, readJson} from '../test-utils.js';

const acceptableTrace = readJson('../fixtures/artifacts/paul/trace.json', import.meta);
const acceptableDevToolsLog = readJson('../fixtures/artifacts/paul/devtoolslog.json', import.meta);

describe('Performance: predictive performance audit', () => {
  // TODO(15841): investigate failures
  if (process.env.INTERNAL_LANTERN_USE_TRACE !== undefined) {
    return;
  }

  it('should compute the predicted values', async () => {
    const artifacts = {
      URL: getURLArtifactFromDevtoolsLog(acceptableDevToolsLog),
      GatherContext: {gatherMode: 'navigation'},
      traces: {
        [PredictivePerf.DEFAULT_PASS]: acceptableTrace,
      },
      devtoolsLogs: {
        [PredictivePerf.DEFAULT_PASS]: acceptableDevToolsLog,
      },
    };
    const context = {computedCache: new Map(), settings: {locale: 'en'}};

    const output = await PredictivePerf.audit(artifacts, context);
    expect(output.displayValue).toBeDisplayString('3,150Â ms');
    const metrics = output.details.items[0];
    for (const [key, value] of Object.entries(metrics)) {
      metrics[key] = value === undefined ? value : Math.round(value);
    }
    expect(metrics).toMatchSnapshot();
  });
});
