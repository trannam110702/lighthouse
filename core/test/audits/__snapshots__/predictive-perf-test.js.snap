// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Performance: predictive performance audit should compute the predicted values 1`] = `
Object {
  "optimisticFCP": 1059,
  "optimisticLCP": 1445,
  "optimisticSI": 261,
  "optimisticTTI": 2132,
  "pessimisticFCP": 1059,
  "pessimisticLCP": 1603,
  "pessimisticSI": 1109,
  "pessimisticTTI": 3981,
  "roughEstimateOfFCP": 1059,
  "roughEstimateOfLCP": 1524,
  "roughEstimateOfLCPLoadEnd": undefined,
  "roughEstimateOfLCPLoadStart": undefined,
  "roughEstimateOfSI": 1059,
  "roughEstimateOfTTFB": 638,
  "roughEstimateOfTTI": 3149,
}
`;
